Assets/Mirror/Components/NetworkSceneChecker.cs

```diff
@@ -0,0 +1,104 @@
+    /// <para>This would be used when the server has multiple additive subscenes loaded to isolate players to their respective subscenes</para>
+    /// </summary>
+    [AddComponentMenu("Network/NetworkSceneChecker")]
+    [RequireComponent(typeof(NetworkIdentity))]
+    [HelpURL("https://mirror-networking.com/docs/Components/NetworkSceneChecker.html")]
+    public class NetworkSceneChecker : NetworkBehaviour
+    {
+        /// <summary>
+        /// How often (in seconds) that this object should update the list of observers that can see it.
+        /// </summary>
+        [Tooltip("How often (in seconds) that this object should check for scene change. Set to zero to disable.")]
+        public float updateInterval = 1;
+
+        /// <summary>
+        /// Flag to force this object to be hidden from all observers.
+        /// <para>If this object is a player object, it will not be hidden for that client.</para>
+        /// </summary>
+        [Tooltip("Enable to force this object to be hidden from all observers.")]
+        public bool forceHidden;
+
+        float lastUpdateTime;
+
+        Scene currentScene;
+
+        void Update()
+        {
+            if (!NetworkServer.active || updateInterval == 0)
+                return;
+
+            if (currentScene == null)
+                currentScene = gameObject.scene;
+
+            if (Time.time - lastUpdateTime > updateInterval)
+            {
+                if (currentScene != gameObject.scene)
+                {
+                    // This object is in a new scene so potential observers in the prior scene
+                    // and the new scene need to rebuild their respective observers lists.
+                    foreach (NetworkIdentity networkIdentity in NetworkIdentity.spawned.Values)
+                    {
+                        Scene objectScene = networkIdentity.gameObject.scene;
+                        if (objectScene == currentScene || objectScene == gameObject.scene)
+                            networkIdentity.RebuildObservers(false);
+                    }
+
+                    currentScene = gameObject.scene;
+                }
+
+                lastUpdateTime = Time.time;
+            }
+        }
+
+        public override bool OnCheckObserver(NetworkConnection conn)
+        {
+            if (forceHidden)
+                return false;
+
+            return (conn.identity.gameObject.scene == gameObject.scene);
+        }
+
+        public override bool OnRebuildObservers(HashSet<NetworkConnection> observers, bool initialize)
+        {
+            // if force hidden then return without adding any observers.
+            if (forceHidden)
+                // always return true when overwriting OnRebuildObservers so that
+                // Mirror knows not to use the built in rebuild method.
+                return true;
+
+            foreach (NetworkConnection conn in NetworkServer.connections.Values)
```

Comment:
this is not very efficient.
unity has several callbacks for when scenes were loaded / changed.
can you try using those instead? for example, have a static dictionary of <scene, hashset<netidentities>>, then onscenechanged(newscene) removes self from old scene, adds self to new scene.

onrebuildobservers simply returns dictionary[scenename].

Reply:
No scenes are being loaded / unloaded.  I'm not aware of any unity callbacks that are raised from `SceneManager.MoveGameObjectToScene`.

Reply:
what if we use update to add a gameobject to dictionary<scene, name> and remove if changed?
check out my networkproximitygridchecker in uMMORPG where I do something similar.
it's way more efficient than brute force

Reply:
See what you think of this version @vis2k